frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/users/UserUtils.java

```diff
@@ -53,68 +61,94 @@ public UserUtils() {
         thriftClients = new ThriftClients();
     }
 
-    public static void synchronizeUserWithDatabase(User user, ThriftClients thriftClients) {
+    public static <T> com.siemens.sw360.datahandler.thrift.users.User synchronizeUserWithDatabase(T source, ThriftClients thriftClients, Supplier<String> emailSupplier, UserSynchronizer<T> synchronizer) {
```

Comment:
There used to be three of them. I've refactored them to only two. By that account, it should have become less confusing.
Anyways, I don't quite see what's confusing about it. Please, put forward your suggestion what method names would clarify the intent for you.